tensorflow/compiler/xla/service/gpu/cudnn_fused_conv_rewriter_test.cc

```diff
@@ -432,6 +441,207 @@ TEST_F(CudnnFusedConvRewriterTest, TestFusedConvClamp) {
+      side_input_f32 = f32[1,3,3,64] convert(side_input)
+
+      conv = s32[1,3,3,64] convolution(inputs32, filters32), window={size=3x3 pad=1_1x1_1}, dim_labels=b01f_01io->b01f, feature_group_count=1
+
+      convfloat = f32[1,3,3,64] convert(conv)
+      scaled_conv = f32[1,3,3,64] multiply(convfloat, alpha_conv)
+      scaled_side_input = f32[1,3,3,64] multiply(side_input_f32, alpha_side_input)
+      broadcasted_bias = f32[1,3,3,64] broadcast(bias), dimensions={3}
+      add1 = f32[1,3,3,64] add(scaled_conv, broadcasted_bias)
+      add2 = f32[1,3,3,64] add(add1, scaled_side_input)
+      relu = f32[1,3,3,64] maximum(zeros, add2)
+
+      lower = f32[] constant(-128)
+      lowers = f32[1,3,3,64] broadcast(lower), dimensions={}
+      upper = f32[] constant(127)
+      uppers = f32[1,3,3,64] broadcast(upper), dimensions={}
+
+      clamp = f32[1,3,3,64] clamp(lowers, relu, uppers)
+
+      ROOT convert = s8[1,3,3,64] convert(clamp) 
+    })");
+}
+
+TEST_F(CudnnFusedConvRewriterTest, TestFusedConvWithScaledFloatSideInputBiasInt8ToInt8) {
+  // From:
+  // convert<int8>(clamp(max(0, alpha_conv * conv(x, w) + alpha_side * float_side_input + bias)));
+  // To:
+  // convert<int8>(clamp(conv(int8_x, int8_w, float_alpha_side, float_side_input, float_bias)));
+  TestMatchWithAllTypes(R"(
+    HloModule Test
+
+    ENTRY Test {
+      zero = f32[] constant(0)
+      zeros = f32[1,3,3,64] broadcast(zero), dimensions={}
+      alpha_conv_scalar = f32[] constant(0.999994934)
+      alpha_conv = f32[1,3,3,64] broadcast(alpha_conv_scalar), dimensions={}
+      alpha_side_input_scalar = f32[] constant(0.899994934)
+      alpha_side_input = f32[1,3,3,64] broadcast(alpha_side_input_scalar), dimensions={}
+
+      input = s8[1,3,3,64] parameter(0)
+      filter = s8[3,3,64,64] parameter(1)
+      side_input = f32[1,3,3,64] parameter(2)
+      bias = f32[64] parameter(3)
+
+      inputs32 = s32[1,3,3,64] convert(input)
+      filters32 = s32[3,3,64,64] convert(filter)
+
+      conv = s32[1,3,3,64] convolution(inputs32, filters32), window={size=3x3 pad=1_1x1_1}, dim_labels=b01f_01io->b01f, feature_group_count=1
+
+      convfloat = f32[1,3,3,64] convert(conv)
+      scaled_conv = f32[1,3,3,64] multiply(convfloat, alpha_conv)
+      scaled_side_input = f32[1,3,3,64] multiply(side_input, alpha_side_input)
+      broadcasted_bias = f32[1,3,3,64] broadcast(bias), dimensions={3}
+      add1 = f32[1,3,3,64] add(scaled_conv, broadcasted_bias)
+      add2 = f32[1,3,3,64] add(add1, scaled_side_input)
+      relu = f32[1,3,3,64] maximum(zeros, add2)
+
+      lower = f32[] constant(-128)
+      lowers = f32[1,3,3,64] broadcast(lower), dimensions={}
+      upper = f32[] constant(127)
+      uppers = f32[1,3,3,64] broadcast(upper), dimensions={}
+
+      clamp = f32[1,3,3,64] clamp(lowers, relu, uppers)
+
+      ROOT convert = s8[1,3,3,64] convert(clamp) 
+    })");
+}
+
+TEST_F(CudnnFusedConvRewriterTest, TestFusedConvWithScaledInt8SideInputBiasInt8ToFloat) {
```

Comment:
No need to have extra check for convert<float>(int8_side_input) in the result HLO, because if it is absorbed by mistake, CuDNN won't run properly and fail the test.